src/components/SkeletonLoading.vue

```diff
@@ -0,0 +1,232 @@
+<template>
+	<div class="placeholder-main" :class="'placeholder-main-' + type">
+		<!-- Placeholder animation -->
+		<template v-for="(suffix, gradientIndex) in ['-regular', '-reverse']">
+			<svg :key="'gradient' + suffix" :class="'placeholder-gradient placeholder-gradient' + suffix">
+				<defs>
+					<linearGradient :id="'placeholder-gradient' + suffix">
+						<stop offset="0%" :stop-color="(gradientIndex === 0) ? colorPlaceholderLight : colorPlaceholderDark" />
+						<stop offset="100%" :stop-color="(gradientIndex === 0) ? colorPlaceholderDark : colorPlaceholderLight" />
+					</linearGradient>
+				</defs>
+			</svg>
+
+			<ul :key="'list' + suffix" :class="'placeholder-list placeholder-list' + suffix">
+				<li v-for="(width, index) in placeholderData" :key="'placeholder' + suffix + index">
+					<svg v-if="type === 'items'"
+						class="items-placeholder"
+						xmlns="http://www.w3.org/2000/svg"
+						:fill="'url(#placeholder-gradient' + suffix + ')'">
+						<circle class="items-placeholder-icon" />
+						<rect class="items-placeholder-line-one" :style="width" />
+					</svg>
+					<svg v-if="type === 'page-heading'"
+						class="page-heading-placeholder"
+						xmlns="http://www.w3.org/2000/svg"
+						:fill="'url(#placeholder-gradient' + suffix + ')'">
+						<circle class="page-heading-placeholder-icon" />
+						<rect class="page-heading-placeholder-line-one" :style="width" />
+						<rect class="page-heading-placeholder-line-two" />
+					</svg>
+					<svg v-if="type === 'text'"
+						class="text-placeholder"
+						xmlns="http://www.w3.org/2000/svg"
+						:fill="'url(#placeholder-gradient' + suffix + ')'">
+						<rect class="text-placeholder-line-one" style="width: 95%" />
```

Comment:
How about using real random computed values instead, like the `width` property of `placeholderData`?